G1ANT.Addon.Selenium/Api/SeleniumWrapper.cs

```diff
@@ -403,6 +404,50 @@ public string GetTextValue(SeleniumCommandArguments search, TimeSpan timeout)
             return res;
         }
 
+        public DataTable GetTableElement(SeleniumCommandArguments search, TimeSpan timeout)
+        {
+            PreCheckCurrentWindowHandle();
+            if (!string.IsNullOrEmpty(search.IFrameSearch?.Value))
+                webDriver.SwitchTo().Frame(FindElement(search.IFrameSearch.Value, search.IFrameBy.Value, timeout));
+            var element = FindElement(search.Search.Value, search.By.Value, timeout);
+
+            if (element.TagName != "table")
+            {
+                try
+                {
+                    element = element.FindElement(By.TagName("table"));
+                }
+                catch (Exception)
+                {
+                    throw new Exception("There are no tables found within the element specified. Try to change the search phrase");
+                }
+            }
+
+            var i = 0;
+            var dataTable = new DataTable();
+            var trElements = element.FindElements(By.TagName("tr"));
+
+            foreach (var trElement in trElements)
+            {
+                while (dataTable.Rows.Count < trElements.Count)
+                    dataTable.Rows.Add();
+
+                var j = 0;
+                var tdElements = trElement.FindElements(By.TagName("td"));
+                foreach (var tdElement in tdElements)
+                {
+                    while (dataTable.Columns.Count < tdElements.Count)
+                        dataTable.Columns.Add();
+
+                    dataTable.Rows[i][j] = tdElement.Text;
+                    j++;
+                }
+                i++;
+            }
```

Comment:
this code block can be simplified:
`
var dataTable = new DataTable();
            var rows = element.FindElements(By.TagName("tr"));

            foreach (var row in rows)
            {
                var cols = row.FindElements(By.TagName("td"));

                while (dataTable.Columns.Count < cols.Count)
                    dataTable.Columns.Add();

                dataTable.Rows.Add(cols.Select(c => c.Text).ToArray());
            }
`

Reply:
I love it